### PR TITLE
Delete time entry state

### PR DESCRIPTION
**Pull request by simonrozsival**
Ref #688 

There will be another PR which will change the behavior of the way `ClientError` transition is handled by the `TogglSyncManagerFactory` to address 4xx errors in compliance with #751 

Squash whenever

### CODE REVIEW

File: Toggl.Foundation.Tests/Sync/States/Push/BaseDeleteEntityStateTests.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        [Fact]
+        public void ReturnsFailTransitionWhenDatabaseOperationFails()
+            => helper.ReturnsFailTransitionWhenDatabaseOperationFails();
+
+        [Fact]
+        public void ReturnsSuccessfulTransitionWhenEverythingWorks()
+            => helper.ReturnsSuccessfulTransitionWhenEverythingWorks();
+
+        [Fact]
+        public void CallsDatabaseDeleteIsWithCorrectParameter()
```
Comment (amulware) [inline Toggl.Foundation.Tests/Sync/States/Push/BaseDeleteEntityStateTests.cs]: Typo? Grammar? Remove 'Is'? (also in the called method)

**Comment by simonrozsival** [comment]
I noticed a problem with deserialization of the "" body returned by the API to the DELETE request which doesn't happen when testing it by running unit tests, but only in simulator. I would like you to look at the two commits I did after your approval.

**Comment by amulware** [comment]
The second commit looks fine. The first confuses me a bit. I didn't realise we were using data sources instead of the database directly anywhere. How come? What is the reasoning here?

I rather have the syncing not depend on the data sources if possible (it's sort of a circular dependency).

**Comment by simonrozsival** [comment]
Well the reasoning is simple - when we use the repository directly, the UI does not update after changes, when we use it, it does. We came to this solution with @willsb and I think it is a good one. Deleting in the datasource however works differently - it only sets the flag. With syncing the repository must be used to work properly, moreover the UI does not need any updating because the time entry should already be hidden (although this does not work right now, but that is a different issue).

**Comment by amulware** [comment]
I see. I originally thought the data sources would subscribe to changes from the sync manager, instead of having dependencies both ways.

I'd still like to go that way at some point, but right now that's not part of this.

Feel free to squash.

**Pull request merged**